warehouse/query-core/src/main/java/datawave/query/iterator/GroupingIterator.java

```diff
@@ -71,27 +75,17 @@ public class GroupingIterator implements Iterator<Map.Entry<Key,Document>> {
     
     private final Iterator<Map.Entry<Key,Document>> previousIterators;
     
-    private final LinkedList<Document> documents = new LinkedList<>();
-    
     Map.Entry<Key,Document> next;
     
-    /**
-     * Length of time in milliseconds that a client will wait for a results to be returned. If a result is not collected before the timeout, a key with an
-     * "intermediate" document will be returned.
-     */
-    private final long resultTimeout;
-    
-    public GroupingIterator(Iterator<Map.Entry<Key,Document>> previousIterators, MarkingFunctions markingFunctions, Collection<String> groupFieldsSet,
-                    int groupFieldsBatchSize, YieldCallback<Key> yieldCallback, long resultTimeout) {
+    public GroupingIterator(Iterator<Map.Entry<Key,Document>> previousIterators, MarkingFunctions markingFunctions, Collection<String> groupFields,
+                    int groupFieldsBatchSize, YieldCallback<Key> yieldCallback, AggregatedFields.Factory aggregateFieldsFactory) {
```

Comment:
Did we lose the functionality that utilizes the resultTimeout?

Reply:
It looks like resultTimeout was added in commit 3b7a68004289971232b7833206156db5b94fa2e4 by Ivan, but no functionality was added that actually used it. 

@ivakegg, was there supposed to be functionality added here for resultTimeout?

Reply:
There is actually a story behind that one.  The grouping mechanism essentially cannot return results to the user until all results on the back end have been gathered.  We were working the ability to allow the iterator stack to return all the way back to the RunningQuery which could then return an empty page to the user and subsequently continue the query with a new next call.  This would prevent the query from timing out.  So to that end this timeout was supposed to work in concert with the timeout in the grouping transform which is used webserver side.
  You will see that the timeout mechanism was actually implemented in the GroupingTransform which is used webserver side.  Then we realized that the RunningQuery could essentially just run the back end "next" call in a separate thread and manage the returning of an empty results page back to the user without having to modify the grouping logic.
  The timeout being passed to the GroupingIterator is a leftover of that.  I was going to go back and actually use that to allow the iterator stack to return and be torn down and rebuilt to avoid scan thread starvation.  However Bill is working that problem in a different way so that parameter can be removed.

Reply:
Thanks for the background info, I'll leave the parameter out.